### PR TITLE
[Chore] Some tweaks to performance

### PR DESCRIPTION
**Pull request by kodster28**
Validation criteria:
- Search still loads on various pages
- Learning path filtering still works on `/learning-paths/`

### CODE REVIEW

File: content/_headers [hunk 1]
```diff
@@ -7,4 +7,5 @@ https://:project.pages.dev/*
   Link: <https://cdn.jsdelivr.net/npm/@docsearch/css@3.5.1>; rel=preload; as=style
   Link: <https://cdn.jsdelivr.net/npm/@docsearch/js@3.5.1>; rel=preload; as=script
   Link: <https://unpkg.com/littlefoot@4.0.0-11/dist/littlefoot.js>; rel=preload; as=script
-  Link: <https://unpkg.com/littlefoot@4.0.0-11/dist/littlefoot.css>; rel=preload; as=style
\ No newline at end of file
+  Link: <https://unpkg.com/littlefoot@4.0.0-11/dist/littlefoot.css>; rel=preload; as=style
+  Link: <https://8MU1G3QO9P-dsn.algolia.net>; rel=preconnect; crossorigin
```
Comment (kodster28) [inline content/_headers:11]: Added in header to align with [Algolia recommendations](https://docsearch.algolia.com/docs/docsearch-v3/#performance-optimization)

File: layouts/partials/head.meta.html [hunk 2]
```diff
@@ -194,8 +194,7 @@
 
   {{- partial "script" (dict "src" "search.ts" "defer" true) -}}
   <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@docsearch/css@3.5.1" />
-  <script src="https://cdn.jsdelivr.net/npm/@docsearch/js@3.5.1"></script>
-  {{- partial "script" (dict "src" "search.ts" "defer" true) -}}
```
Comment (kodster28) [inline layouts/partials/head.meta.html:198]: Remove double invocation of script.

File: layouts/partials/head.meta.html [hunk 3]
```diff
@@ -223,7 +222,6 @@
 
   <script src="{{ $script.RelPermalink }}" defer data-token="{{ $coveo.token }}" data-org="{{ $coveo.org }}"></script>
   {{- else if (and $coveo (eq $.Context.RelPermalink "/tutorials/")) -}}
-  {{- partial "script" (dict "src" "search.redirect.ts" "defer" true) -}}
```
Comment (kodster28) [inline layouts/partials/head.meta.html:226]: Remove script that doesn't exist any more.

File: layouts/partials/head.meta.html [hunk 4]
```diff
@@ -194,8 +194,7 @@
 
   {{- partial "script" (dict "src" "search.ts" "defer" true) -}}
   <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@docsearch/css@3.5.1" />
-  <script src="https://cdn.jsdelivr.net/npm/@docsearch/js@3.5.1"></script>
-  {{- partial "script" (dict "src" "search.ts" "defer" true) -}}
+  <script defer async src="https://cdn.jsdelivr.net/npm/@docsearch/js@3.5.1"></script>
```
Comment (kodster28) [inline layouts/partials/head.meta.html:197]: Updated properties to async defer load

**Review by haleycode — APPROVED** [review]

**Pull request merged**